subprojects/core/src/main/java/org/gradle/api/internal/file/copy/NormalizingCopyActionDecorator.java

```diff
@@ -142,112 +143,138 @@ public long getLastModified() {
 
         @Override
         public long getSize() {
-            throw new UnsupportedOperationException();
+            throw new UnsupportedOperationException("It's a synthetic FileCopyDetails just to create parent directories");
         }
 
         @Override
         public InputStream open() {
-            throw new UnsupportedOperationException();
+            throw new UnsupportedOperationException("It's a synthetic FileCopyDetails just to create parent directories");
         }
 
         @Override
         public RelativePath getRelativePath() {
             return path;
         }
 
+        @Override
+        public boolean copyTo(File target) {
```

Comment:
:thinking: isn't base class' implementation enough? It looks to me that it does the right thing with the proper  `getPermissions` override.

Reply:
It is. However, in https://github.com/gradle/gradle/pull/25264 I plan to deprecate `AbstractFileTree.copyTo` since it's one-size-fits-all and unsuitable for cases with symlinks. On top of that, the implementation here is a bit simpler and clarifies the purpose of creating a directory.